lib/most_frequent_integer.rb

```diff
@@ -3,5 +3,52 @@
 # exists, return nil.
 
 def find_most_frequent_integer(arr)
-  # put your code here
+
+sortarr = arr.sort
+indexarr = Array.new
+n = 0
+
+if sortarr.empty?
+    return nil
+else
+    i = sortarr[0]
+end
+
+while n < sortarr.length && i <= sortarr.last
+
+    indexarr[n] = sortarr.each_index.select{|x| sortarr[x] == i} 
+    i = i + 1
+    n = n + 1
+    
 end
+
+m = 0
```

Comment:
I was having issues with infinite loops. I had used m to stop that.

My scribble page was running out of space and single letter variables helped save enough space. When I get a coding job I will have enough money to get two scribble pages!